File: solutions/typescript/wordy/1/wordy.ts
export const answer = (question: string): number => {
  if (typeof question !== 'string') throw new Error('Unknown operation');

  // Ensure format
  if (!question.startsWith('What is') || !question.endsWith('?')) {
    throw new Error('Unknown operation');
  }

  // Clean and prepare the string
  let expression = question
    .replace('What is', '')
    .replace('?', '')
    .trim();

  // Replace verbal math operations
  expression = expression
    .replace(/plus/g, '+')
    .replace(/minus/g, '-')
    .replace(/multiplied by/g, '*')
    .replace(/divided by/g, '/');

  // Split into tokens
  const tokens = expression.split(' ').filter(Boolean);

  // Validate tokens
  const validTokens = /^-?\d+$|[+\-*/]$/;
  if (!tokens.every((t) => validTokens.test(t))) {
    throw new Error('Unknown operation');
  }

  // Must start with a number
  if (isNaN(parseInt(tokens[0]))) {
    throw new Error('Syntax error');
  }

  // Evaluate left-to-right (no operator precedence)
  let result = parseInt(tokens[0]);

  for (let i = 1; i < tokens.length; i += 2) {
    const operator = tokens[i];
    const operand = parseInt(tokens[i + 1]);

    if (isNaN(operand)) throw new Error('Syntax error');

    switch (operator) {
      case '+':
        result += operand;
        break;
      case '-':
        result -= operand;
        break;
      case '*':
        result *= operand;
        break;
      case '/':
        result /= operand;
        break;
      default:
        throw new Error('Unknown operation');
    }
  }

  return result;
};

File: solutions/go/wordy/1/wordy.go
package wordy

import (
	"strconv"
	"strings"
)

// Answer parses and evaluates simple math word problems.
func Answer(question string) (int, bool) {
	// Trim question marks and extra spaces
	q := strings.TrimSpace(question)
	q = strings.TrimSuffix(q, "?")

	// Must start with "What is "
	if !strings.HasPrefix(q, "What is ") {
		return 0, false
	}

	// Remove the starting phrase
	q = strings.TrimPrefix(q, "What is ")

	// Replace verbal operators with symbols for clarity
	q = strings.ReplaceAll(q, "multiplied by", "mul")
	q = strings.ReplaceAll(q, "divided by", "div")
	q = strings.ReplaceAll(q, "plus", "add")
	q = strings.ReplaceAll(q, "minus", "sub")

	// Split into words
	tokens := strings.Fields(q)
	if len(tokens) == 0 {
		return 0, false
	}

	// Try parsing the first number
	current, err := strconv.Atoi(tokens[0])
	if err != nil {
		return 0, false
	}

	i := 1
	for i < len(tokens) {
		if i+1 >= len(tokens) {
			return 0, false // incomplete operation
		}

		op := tokens[i]
		numStr := tokens[i+1]
		num, err := strconv.Atoi(numStr)
		if err != nil {
			return 0, false
		}

		switch op {
		case "add":
			current += num
		case "sub":
			current -= num
		case "mul":
			current *= num
		case "div":
			current /= num
		default:
			return 0, false // unsupported operation
		}
		i += 2
	}

	return current, true
}
